tests/argocd_external_link/00-assert.yaml

```diff
@@ -0,0 +1,21 @@
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: argocd-external-link
+  annotations:
+    "link.argocd.argoproj.io/external-link": "https://example.com"
+spec:
+  selector:
+    matchLabels:
+      app: argocd-external-link
+  template:
+    metadata:
+      annotations:
+        prometheus.io/scrape: "true"
+      labels:
+        app: argocd-external-link
+    spec:
+      containers:
+        - name: argocd-external-link
+          image: image
```

Comment:
when asserting using ´kuttl´ you only need to check for what you are adding. in this case the spec of the Deployment is not really interesting, but whether or not the annotation for argo is. i would remove the spec entirely.